src/controller.js

```diff
@@ -120,14 +119,24 @@ export class Controller {
       observer.selfSubscriber = selfSubscriber;
     }
 
+    let overrideContext;
     if (this.view !== null) {
       if (skipSelfSubscriber) {
         this.view.viewModelScope = scope;
       }
-
-      this.view.bind(this.viewModel, createOverrideContext(this.viewModel, scope.overrideContext));
+      if (this.viewModel === scope.overrideContext.bindingContext) {
+        overrideContext = scope.overrideContext;
+      } else if (this.instruction.inheritBindingContext) {
+        overrideContext = createOverrideContext(this.viewModel, scope.overrideContext);
+      } else {
+        overrideContext = createOverrideContext(this.viewModel);
+        overrideContext.__parentOverrideContext = scope.overrideContext;
+      }
+      this.view.bind(this.viewModel, overrideContext);
     } else if (skipSelfSubscriber) {
-      this.viewModel.bind(context, scope.overrideContext);
+      overrideContext = Object.assign({}, scope.overrideContext);
+      overrideContext.parentOverrideContext = scope.overrideContext.__parentOverrideContext;
```

Comment:
clone the overrideContext so we can hook up the parent scope without impacting other views/controllers that are using this overrideContext instance